src/test/java/com/example/report/controller/ReportControllerTest.java

```diff
@@ -145,4 +150,16 @@ void getReportEntriesByReportId() throws Exception {
         assertThat(reportEntry.getNetAmount()).isEqualByComparingTo(actualReportEntryDTOsList.get(0).getNetAmount());
         assertThat(reportEntry.getPensionsFund()).isEqualByComparingTo(actualReportEntryDTOsList.get(0).getPensionsFund());
     }
+
+
+    @Test
+    void extractAllReports() throws Exception {
+        LocalDate testDate = LocalDate.of(2022, 8, 15);
+        Report report1 = new Report(10L, testDate, testDate);
+        Report report2 = new Report(16L, testDate, testDate);
+        reportRepository.saveAll(List.of(report1, report2));
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/report/extractReports"))
+                .andExpect(status().isCreated());
```

Comment:
We can try parsing the response into bytes then into input stream and then into workbook to check the data inside.
`String contentAsString = mockMvc.perform(MockMvcRequestBuilders.get("/report/extractReports"))
                .andExpect(status().isCreated())
                .andReturn().getResponse().getContentAsString();

        byte[] contentAsStringBytes = contentAsString.getBytes();

        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(contentAsStringBytes);

        Workbook workbook = WorkbookFactory.create(byteArrayInputStream);`